src/cpp/server/secure_server_credentials.h

```diff
@@ -72,10 +72,8 @@ class SecureServerCredentials final : public ServerCredentials {
   void SetAuthMetadataProcessor(
       const std::shared_ptr<grpc::AuthMetadataProcessor>& processor) override;
 
-  grpc_server_credentials* c_creds() { return creds_; }
-
  private:
-  SecureServerCredentials* AsSecureServerCredentials() override { return this; }
+  grpc_server_credentials* c_creds() const override { return creds_; }
```

Comment:
I think the `SecureServerCredentials` class can go away.

Reply:
As discussed, we were able to reduce the footprint, but making it go away was not necessarily the right thing to do just yet.